src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -263,8 +263,15 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         });
 
         // extract fragment
-        $fragment = isset($extra['_fragment']) ? $extra['_fragment'] : '';
-        unset($extra['_fragment']);
+        $fragment = '';
+        if (array_key_exists('_fragment', $defaults)) {
+            $fragment = $defaults['_fragment'];
+        }
+
+        if (array_key_exists('_fragment', $extra)) {
```

Comment:
You changed behaviour here for the case when `_fragment` is `null`. [`isset()` will return false, while `array_key_exists()` will return true for a key with `null` value](https://3v4l.org/eRKSS).